### PR TITLE
"Rewrite" using Gatsby 5 and TypeScript

### PR DESCRIPTION
**Pull request by puls**
One way to get everything up to date and out of dependency hell is to start a fresh project and then bring over the old source code as straightforwardly as possible.

This PR does just that.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,102 @@
+{
+  "compilerOptions": {
```
Comment (jonahgreenthal) [inline tsconfig.json:2]: Do we want to keep all the commented-out stuff in this file?

File: package.json [hunk 2]
```diff
@@ -1,55 +1,38 @@
 {
   "name": "qb-schema",
+  "version": "2.1",
```
Comment (jonahgreenthal) [inline package.json:3]: We should probably bump the version number (per [this comment](https://github.com/quizbowl/schema/pull/80#discussion_r289256718)).

**Review by jonahgreenthal — CHANGES_REQUESTED** [review]
Fundamentally, this works fine for me.

However, changes to the `.graphql` files are no longer automatically listened to, I have to restart the dev server to pick them up. If that's easy to fix, great, but it's not worth spending time on.

Also, I get some npm warnings. I don't know if they matter.
```
$ npm install
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: react-server-dom-webpack@0.0.0-experimental-c8b778b7f-20220825
npm WARN Found: react@18.2.0
npm WARN node_modules/react
npm WARN   react@"^18.2.0" from the root project
npm WARN   11 more (@gatsbyjs/reach-router, @mdx-js/react, gatsby, ...)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer react@"0.0.0-experimental-c8b778b7f-20220825" from react-server-dom-webpack@0.0.0-experimental-c8b778b7f-20220825
npm WARN node_modules/react-server-dom-webpack
npm WARN   react-server-dom-webpack@"0.0.0-experimental-c8b778b7f-20220825" from gatsby@5.4.0
npm WARN   node_modules/gatsby
npm WARN
npm WARN Conflicting peer dependency: react@0.0.0-experimental-c8b778b7f-20220825
npm WARN node_modules/react
npm WARN   peer react@"0.0.0-experimental-c8b778b7f-20220825" from react-server-dom-webpack@0.0.0-experimental-c8b778b7f-20220825
npm WARN   node_modules/react-server-dom-webpack
npm WARN     react-server-dom-webpack@"0.0.0-experimental-c8b778b7f-20220825" from gatsby@5.4.0
npm WARN     node_modules/gatsby
```

**Comment by puls** [comment]
https://github.com/gatsbyjs/gatsby/issues/37242#issuecomment-1348142685

File: README.md [hunk 3]
```diff
@@ -1,46 +1,61 @@
...
 
-    Navigate into your new site’s directory and start it up.
+    Navigate into the site’s directory and start it up.
 
-    ```sh
-    cd path-to-qb-schema/
-    yarn && yarn dev
+    ```shell
+    cd schema/
+    npm run develop
```
Comment (jonahgreenthal) [inline README.md:9]: Probably need to `npm install`, right?

**Review by jonahgreenthal — APPROVED** [review]
Nice!

**Pull request merged**